drgn_tools/lsmod.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+import argparse
+from typing import Iterable
+from typing import List
+
+from _drgn import FaultError
+from drgn import cast
+from drgn import Object
+from drgn import Program
+from drgn.helpers.common.format import escape_ascii_string
+from drgn.helpers.linux.list import list_for_each_entry
+
+from drgn_tools.corelens import CorelensModule
+from drgn_tools.module import KernelModule
+from drgn_tools.table import print_table
+
+
+def for_each_module_use(
+    prog: Program, source_list_addr: Object
+) -> Iterable[Object]:
+    """
+    Provide the list of ``struct module_use`` as an iterable object
+
+    :param source_list_addr: ``struct module_use.source_list `` Object.
+    :returns: A list of ``struct module.source_list`` as an iterable object
+    """
+    return list_for_each_entry(
+        "struct module_use", source_list_addr, "source_list"
+    )
+
+
+def print_module_parameters(prog: Program) -> None:
+    """
+    Provide the list of ``module parameters`` as an iterable object
+
+    :param prog: drgn program.
+    :returns: None
+    """
+
+    # Module details
+    for mod in KernelModule.all(prog):
+        print("\n")
+        print("MODULE NAME:".ljust(15), mod.name)
+        print("PARAM COUNT:", str(mod.obj.num_kp.value_()))
+        print("ADDRESS    :", hex(mod.obj.num_kp.address_of_()))
+        table_value = []
+        ker_parm = hex(mod.obj.kp.value_())
+        table_value.append(["PARAMETER", "ADDRESS", "TYPE", "VALUE"])
+        integer_params = {
+            "byte": "unsigned char *",
+            "short": "short *",
+            "ushort": "unsigned short *",
+            "int": "int *",
+            "uint": "unsigned int *",
+            "long": "long *",
+            "ulong": "unsigned long *",
+        }
+        for i in range(0, mod.obj.num_kp):
+            parm_valf = ""
+            parm_val = ""
+            kp = mod.obj.kp[i]
+            try:
+                param_type = prog.symbol(kp.ops.get).name
+            except LookupError:
+                continue
+
+            if param_type.startswith("param_get_"):
+                param_type = param_type[len("param_get_") :]
+                if param_type in integer_params:
+                    parm_valf_address = cast(
+                        integer_params[param_type], kp.arg
+                    )
+                    try:
+                        parm_valf = parm_valf_address[0].value_()
+                    except FaultError:
+                        parm_valf = "(page fault)"
+                        continue
```

Comment:
Did you encounter page faults with integer types? I wouldn't expect that. If you're not working around a known issue, I'd prefer you not catch this error.

Regardless, with the `continue` statement here, the parameter will get skipped and won't be added to the list in the `FaultError` case. We should be including all parameters even on error. It would be best to remove the `continue` statement.

Reply:
Yes we have encountered and below is the localtest done on the vmcore 
[apalakun@lse-india-drgn drgn-tools]$ python3 -m drgn_tools.corelens /drgn_vmcore/githubCI/rds-uek5/vmcore --debuginfo /drgn_vmcore/githubCI/rds-uek5/ -M lsmod > ~/4.txt
error: Encountered exception in lsmod:
Traceback (most recent call last):
  File "/drgn_home/apalakun/github/drgn-tools/drgn_tools/corelens.py", line 340, in _run_module
    mod.run(prog, args)
  File "/drgn_home/apalakun/github/drgn-tools/drgn_tools/lsmod.py", line 157, in run
    print_module_parameters(prog)
  File "/drgn_home/apalakun/github/drgn-tools/drgn_tools/lsmod.py", line 72, in print_module_parameters
    parm_valf = parm_valf_address[0].value_()
_drgn.FaultError: could not read memory from kdump: Cannot get page I/O address: Page not present: pte[477] = 0x0: 0xffffffffc05dd000